src/memory/memoryManager.test.ts

```diff
@@ -26,39 +26,33 @@ describe('MemoryManager', () => {
       const memoryManager = new MemorySegmentManager();
       memoryManager.addSegment();
 
-      expect(memoryManager.memory.getNumSegments()).toEqual(
-        UnsignedInteger.toUint32(1).value as Uint32
-      );
+      expect(memoryManager.memory.getNumSegments()).toEqual(1);
     });
   });
-  describe('loadData', () => {
+  describe('setData', () => {
     test('should return the final state of the pointer', () => {
       const memoryManager = new MemorySegmentManager();
       memoryManager.addSegment();
       const address = new Relocatable(0, 0);
-      const { value: loadedAddress } = memoryManager.loadData(address, DATA);
+      const newAddress = memoryManager.setData(address, DATA);
```

Comment:
I know that even though I don't like it, js does return the latest value when doing `arr.push(1, 2, 3)` (ie.

```
> let a = []
undefined
> a.push(1, 2)
2
> a
[ 1, 2 ]
```

but here it's returning the next free memory pointer, do we really want this? When I read something like `this.mutateWith(args0, args1)` I don't expect a returned value